sklearn/gaussian_process/tests/test_kernels.py

```diff
@@ -89,39 +91,68 @@ def test_kernel_theta():
 
         # Check that values returned in theta are consistent with
         # hyperparameter values (being their logarithms)
-        for i, hyperparameter in enumerate(kernel.hyperparameters):
-            assert_equal(theta[i],
-                         np.log(getattr(kernel, hyperparameter.name)))
+        i = 0
```

Comment:
Yes, that's what prompted me to check  further as to how did RBF do through and then encountered that they are being ignored. Nevertheless, removing the lines causes a fail:
```python
AssertionError: Items in the first set but not the second:
'k1__constant_value'
'k2__length_scale'
    """Fail immediately, with the given message."""
>>  raise self.failureException("Items in the first set but not the second:\n'k1__constant_value'\n'k2__length_scale'")
```
I'll look into the issue and try to resolve it so that we can run the tests without ignoring any.